openbas-api/src/test/java/io/openbas/rest/ScenarioApiTest.java

```diff
@@ -0,0 +1,322 @@
+  @Mock
+  private InjectRepository injectRepository;
+  private InjectService injectService;
+  @Mock
+  private ScenarioImportApi scenarioImportApi;
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  File testFile;
+
+  static String SCENARIO_ID;
+
+  static Scenario mockedScenario;
+
+  static ImportMapper mockedImportMapper;
+
+  @BeforeEach
+  void before() throws FileNotFoundException {
+    // Getting a test file
+    testFile = ResourceUtils.getFile("classpath:xls-test-files/test_file_1.xlsx");
+
+    // Creating a mocked scenario
+    SCENARIO_ID = UUID.randomUUID().toString();
+    mockedScenario = new Scenario();
+    mockedScenario.setId(SCENARIO_ID);
+
+    // Creating a mocked Import Mapper
+    UUID importMapperId = UUID.randomUUID();
+    mockedImportMapper = createImportMapper(importMapperId.toString());
+    injectService = Mockito.mock(InjectService.class);
+    // Injecting mocks into the controller
+    scenarioImportApi = new ScenarioImportApi(scenarioRepository, injectRepository, importMapperRepository, injectService);
+
+    mvc = MockMvcBuilders.standaloneSetup(scenarioImportApi)
+            .build();
+  }
+
+  // -- SCENARIOS --
+
+  @DisplayName("Post an XLS file")
+  @Test
+  void postAnXLSFile() throws Exception {
+    // -- PREPARE --
+    InputStream in = new FileInputStream(testFile);
+    MockMultipartFile xlsFile = new MockMultipartFile("file",
+            "my-awesome-file.xls",
+            "application/xlsx",
+            in.readAllBytes());
+    // -- EXECUTE --
+    String response = this.mvc
+        .perform(MockMvcRequestBuilders.multipart(SCENARIO_URI + "/" + SCENARIO_ID + "/xls")
+                .file(xlsFile))
+        .andExpect(status().is2xxSuccessful())
+        .andReturn()
+        .getResponse()
+        .getContentAsString();
+
+    // -- ASSERT --
+    assertNotNull(response);
+
+    ImportPostSummary returnValue = objectMapper.readValue(response, ImportPostSummary.class);
+    try {
+      UUID.fromString(returnValue.getImportId());
+    } catch (Exception ex) {
+      fail();
+    }
+    assertEquals(1, returnValue.getAvailableSheets().size());
+    assertEquals("CHECKLIST", returnValue.getAvailableSheets().get(0));
+  }
+
+  @DisplayName("Post a corrupted XLS file")
+  @Test
+  void postACorruptedXLSFile() throws Exception {
+    // -- PREPARE --
+    InputStream in = new FileInputStream(testFile);
+    MockMultipartFile xlsFile = new CustomMockMultipartFile("file",
+            "my-awesome-file.xls",
+            "application/xlsx",
+            in.readAllBytes());
+
+    // -- EXECUTE --
+    this.mvc
+      .perform(MockMvcRequestBuilders.multipart(SCENARIO_URI + "/" + SCENARIO_ID + "/xls")
+              .file(xlsFile))
+      .andExpect(status().is4xxClientError())
+      .andExpect(result -> {assertTrue(result.getResolvedException() instanceof BadRequestException);});
+  }
+
+  @DisplayName("Post and test XLS file")
+  @Test
+  void postAndTestXLSFile() throws Exception {
```

Comment:
As discussed together, since we're doing unit tests, we're only testing the api, not the service itself. The unit tests for the service are done in another test